src/accessors.d

```diff
@@ -84,24 +84,24 @@ template GenerateReader(string name, alias field)
     static enum helper()
     {
         import std.string : format;
-        import std.traits : ForeachType, isArray, isSomeString, MutableOf;
+        import std.traits : isArray, isSomeString;
 
         enum visibility = getVisibility!(field, Read);
-        enum outputType = typeName!(typeof(field));
         enum accessorName = accessor(name);
 
         static if (isArray!(typeof(field)) && !isSomeString!(typeof(field)))
         {
-            enum valueType = typeName!(MutableOf!(ForeachType!(typeof(field))));
-
-            return format("%s final @property inout(%s)[] %s() inout " ~
-                "{ inout(%s)[] result = null; result ~= this.%s; return result; }",
-                visibility, valueType, accessorName, valueType, name);
+            return format("%s final @property auto %s() inout {"
+                        ~ "import std.traits;"
+                        ~ "inout(ForeachType!(typeof(this.%s)))[] result = null;"
+                        ~ "return result ~ this.%s;"
```

Comment:
Is there a more explicit way to copy the array?

Reply:
I'm aware only of `.dup`.

Reply:
try `.dup`; otherwise, try `[] ~ ...`

Reply:
.dup:
```
src/accessors.d-mixin-666-mixin-668(668,58): Error: template object.dup cannot deduce function from argument types !()(inout(const(X)[])), candidates are:
/usr/include/dmd/druntime/import/object.d(1943,6):        object.dup(T : V[K], K, V)(T aa)
/usr/include/dmd/druntime/import/object.d(1979,6):        object.dup(T : V[K], K, V)(T* aa)
/usr/include/dmd/druntime/import/object.d(3737,16):        object.dup(T)(T[] a) if (!is(const(T) : T))
/usr/include/dmd/druntime/import/object.d(3753,15):        object.dup(T)(const(T)[] a) if (is(const(T) : T))
src/accessors.d(677,9): Error: static assert  is(typeof(y) == const(X)[]) is false
```